iphone/Classes/TiLocale.m

```diff
@@ -44,6 +44,11 @@ +(NSString*)defaultLocale
 +(void)setLocale:(NSString*)locale
 {
 	TiLocale *l = [TiLocale instance];
+	NSString *var = [[UIDevice currentDevice] systemVersion];
+	if (var.floatValue >= 9.0) {
+		//[TIMOB-19566]:Truncate the current locale for parity between iOS versions
+		locale =[locale substringToIndex:2];
+	}
 	l.currentLocale = locale;
 	NSString *path = [[ NSBundle mainBundle ] pathForResource:locale ofType:@"lproj" ];
```

Comment:
'locale' is used here for referencing a path, so i'm not sure it's a good idea to change this variable, since the ticket wants to address currentLanguage only. Considering the return value of [TiLocal defaultLocale] is only used for getCurrentLanguage, it maybe a better idea to change the make this correction here instead https://github.com/AngelkPetkov/titanium_mobile/blob/TIMOB-19566-currentLanguage/iphone/Classes/TiLocale.m#L41